shared/editor/embeds/Grist.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from "react";
+import Frame from "../components/Frame";
+import Image from "../components/Image";
+import { EmbedProps as Props } from ".";
+
+function Grist(props: Props) {
+  const { matches } = props.attrs;
+  const sheetURL = matches[1];
+  return (
+    <Frame
+      {...props}
+      src={`${sheetURL}?embed=true`}
+      icon={
+        <Image
+          src="/images/grist.png"
+          alt="Grist Icon"
+          width={16}
+          height={16}
+        />
+      }
+      title="Grist Spreadsheet"
+      border
+    />
+  );
+}
+
+Grist.ENABLED = [
+  new RegExp(
+    "^(https?://[a-z.-]+\\.?getgrist\\.com/.*/p/[0-9]+)(\\?embed=true)?$"
```

Comment:
Couple of concerns here...

1. It matches `https://fakegetgrist.com/doc/fake/p/1`, we surely don't want this :)
2. Maybe the part `/p/[0-9]+` is too strict. The [`src`](https://github.com/gristlabs/grist-core/blob/9434b8e4549713212c1ad5b0796a2e9db2511cd6/app/common/gristUrls.ts#L174) seems to suggest that `p` is followed by `docPage`, which in itself could be a [number or a string](https://github.com/gristlabs/grist-core/blob/9434b8e4549713212c1ad5b0796a2e9db2511cd6/app/common/gristUrls.ts#L82). Moreover, seems like `p/<docPage` is optional.

I'd suggest
```suggestion
    "^https?://([a-z.-]+\\.)?getgrist\\.com/doc/(.+)$"
```